Cargo.toml

```diff
@@ -17,13 +17,15 @@ edition = "2018"
 [dependencies]
 byteorder = { version = "1", default-features = false }
 ff = { version = "0.7", default-features = false }
-rand = { version = "0.7", default-features = false }
-rand_xorshift = "0.2"
+rand = { version = "0.7", optional = true, default-features = false }
+rand_core = { version = "0.5", default-features = false }
+rand_xorshift = { version = "0.2", optional = true }
 subtle = { version = "2.2.1", default-features = false }
 
 [features]
-default = ["alloc"]
+default = ["tests"]
```

Comment:
I think we can probably reduce this to `alloc`; the generic tests are available for downstream implementors of the `group` traits to leverage if they so wish, but I don't think it needs to be a default requirement for all consumers. It's somewhat unfortunate that [it isn't yet possible to automatically activate features for tests](https://github.com/rust-lang/cargo/issues/2911), so a downstream crate using the tests likely means that the feature is enabled for everyone anyway, but let's try to minimise if we can.

Reply:
Done. Since `tests` is the only thing that depends on `alloc` though really, we might consider simply removing the `default` line entirely.

Reply:
`Wnaf` (which is used by `bellman`) also depends on `alloc`. That being said, I know that the ecosystem convention is for crates with a `std` feature flag to default-depend on it, but I'm not familiar with the convention for crates with only an `alloc` flag.

Reply:
Aah ok, well this seems fine to me then!